### PR TITLE
Getting started with python plug-ins for GIMP

### LINKED ISSUE #4843: [python]Getting started with python plug-ins for GIMP
#### Proposed title of the article
Getting started with python plug-ins for GIMP
#### Proposed article introduction
GIMP is an open-source picture control program that many individuals use as a feasible option in contrast to a portion of the business contributions. GIMP handles convoluted components like layers and ways. GIMP upholds various diverse picture organizations and accompanies generally complex channels. GIMP has solid local area backing and inclusion, so it is generally simple to track down data concerning how to utilize or expand GIMP.
This article will turn out how to utilize GIMP-Python, which is an assortment of Python modules that permit you to program in Python to computerize errands in the GNU Image Manipulation Program (GIMP). These Python modules are basically coverings for the libgimp library. GIMP-Python isn't as old as Script-Fu expansions. To run scripts in Script-Fu, a module is utilized. The Python script becomes the dominant focal point in GIMP-Python and behaviors the work. You might run the GIMP-Python scripts straightforwardly from GIMP, or you can utilize GIMP's clump mode to begin it from the order line. 

#### Key takeaways 
- Enrolling your content
- Scripting the resize
- Running the resize module script
- Prearranging the picture change
- Running the refreshed module script
- Running both on a folder
#### Article quality
In this article, you will figure out how to foster Python code to computerize two unique cycles in GIMP: picture resizing and document saving. It will likewise help the student on the best way to use GIMP-python.
#### References
N/A

### PR DESCRIPTION
**Pull request by kmwangi-ken**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4843

### CODE REVIEW

File: content/articles/getting-started-with-python-plug-ins-for-gimp/index.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+### Introduction 
+GIMP is an open-source image manipulation tool which many people use as a suitable solution as it competes with some corporate demands. GIMP is efficient in dealing with tangled elements such as layers and paths. GIMP is aware of many coordinated picture associations and works with channels that are typically elaborate. GIMP has a solid foundation and philosophy, thus there isn't any technical info as to how to use or encourage GIMP in general.
```
Comment (mercymeave) [inline content/articles/getting-started-with-python-plug-ins-for-gimp/index.md:2]: This sentence is too long. What are tangled elements? 

When you say `GIMP is aware of many coordinated picture associations and works with channels that are typically elaborate. GIMP has a solid foundation and philosophy, thus there isn't any technical info as to how to use or encourage GIMP in general.` what do you mean?

File: content/articles/getting-started-with-python-plug-ins-for-gimp/index.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
+### Introduction 
+GIMP is an open-source image manipulation tool which many people use as a suitable solution as it competes with some corporate demands. GIMP is efficient in dealing with tangled elements such as layers and paths. GIMP is aware of many coordinated picture associations and works with channels that are typically elaborate. GIMP has a solid foundation and philosophy, thus there isn't any technical info as to how to use or encourage GIMP in general.
+This article will wind up being about how to utilize GIMP-Python, which is a party of Python modules that permit you to program in Python to mechanize undertakings in the GNU Image Manipulation Program (GIMP). These Python modules are generally covered for the libgimp library. GIMP-Python isn't generally so old as Script-Fu upgrades. To run scripts in Script-Fu, a module is used. The Python script changes into the overpowering characteristic of gathering in GIMP-Python and practices the work. You may run the GIMP-Python scripts clearly from GIMP, or you can use GIMP's group mode to start it from the sales line.
```
Comment (mercymeave) [inline content/articles/getting-started-with-python-plug-ins-for-gimp/index.md:3]: Your sentences are too long and wordy.

Comment (mercymeave) [inline content/articles/getting-started-with-python-plug-ins-for-gimp/index.md:3]: Avoid mixing up ideas.

File: content/articles/getting-started-with-python-plug-ins-for-gimp/index.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+- [Running the resize module script](#running-the-resize-module-script)
+- [Setting up the picture change](#setting-up-the-picture-change)
+- [Running the reestablished module script](#running-the-reestablished-module-script)
+- [Running both on a folder](#running-both-on-a-folder)
+- [Conclusion](#conclusion)
+- [Further reading](#further-reading)
+
+### Installation 
+Both gimp and python can be installed and used at any time in `Mac OS`, `Linux`, and `Microsoft Windows`. You can use Python to create advanced modules for GIMP and run them on these stages.
+GIMP is an open-source image editing tool that many folks use as a personal choice rather than as part of professional commitments, making it simpler to show on your device. You can get the most recent version of GIMP [here](http://www.gimp.org/downloads/).
```
Comment (mercymeave) [inline content/articles/getting-started-with-python-plug-ins-for-gimp/index.md:19]: You are repeating the definition of GIMP.

**Review by mercymeave — COMMENTED** [review]
Hello, there seem to be a lot of readability issues with your article, kindly go over the published articles to master the simplicity and format of polished articles.  Your article seems to have a lot of words but does not provide any flow of content that the reader can understand.

I haven't gone through the entire article so be sure to have a polished article. I will be closing this PR to give you more time to work on it. Not  a problem just create a new PR  after you are done. 

Closing a PR does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.

**Comment by kmwangi-ken** [comment]
Hi @mercymeave, I have just made the requested changes and counter-checked throughout the article to reduce the review complexity. I was requesting if you could reopen the PR so that I could push the change. Thank you.

**Comment by kmwangi-ken** [comment]
@mercymeave, or should I open a new PR?

**Comment by mercymeave** [comment]
Yes please open a new PR and ensure you have worked on all the readability issues to avoid closure of the article.

**Pull request closed without merging**